test/test_misc.F90

```diff
@@ -9,7 +9,11 @@ program test_misc
   integer lun, il, im
   integer ios
   integer num, iret
-
+  integer ireadmm, imsg, idate, iunit
+  integer mbay(2)
```

Comment:
Similar to the above comment, the ireadmm function should be explicitly declared as integer*4 here, since the below ireadmm test isn't encased in an `#ifdef KIND_4` block and therefore runs for all of the _4, _8, and _d tests.

Contrast this with the igetprm function declaration in line 24, which is encased in an `#ifdef KIND_4` block and therefore only ever runs for the _4 test, and so for that reason it's OK to just declare that one as integer.